client/controller/web-util.js

```diff
@@ -1,9 +1,10 @@
-import ReactDom from 'react-dom';
+import { hydrateRoot } from 'react-dom/client';
+import { getRootNode, setRootNode, getRootDomElement } from 'calypso/boot/common';
```

Comment:
The imports should be the other way around. The `boot/common` module should import the `render` function from `client/controller` and use it. The `wpcomRootNode` singleton should be maintained inside `web-utils.js` and can be private to the module. The first call to `render` or `hydrate` will initialized the root object.

That should solve all the problems with exposing `setRootNode` etc.